### PR TITLE
Provide endpoint to resource creation methods

### PR DESCRIPTION
**Pull request by Marlinc**
This makes it (for example) possible to access the schema when creating resources.

### CODE REVIEW

File: src/Webservice/Webservice.php [hunk 1]
```diff
@@ -282,15 +283,15 @@ protected function _logQuery(Query $query, LoggerInterface $logger)
      * Loops through the results and turns them into resource objects
      *
      * @param array $results Array of results from the API
-     * @param string $resourceClass The resource class to use
+     * @param Endpoint $endpoint The endpoint class to use
```
Comment (ADmad) [inline src/Webservice/Webservice.php]: Use FQCN

**Comment by ADmad** [comment]
Tests failing. Can you please also replace coveralls with codecov. You can check foc/search as reference for changes to .travis.yml and phpunit.xml.dist.

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 language: php
 
 php:
-  - 5.4
```
Comment (ADmad) [inline .travis.yml]: Need to test on 5.4 since we support 3.0+

File: .travis.yml [hunk 3]
```diff
@@ -20,11 +19,11 @@ matrix:
   fast_finish: true
 
   include:
-    - php: 5.4
+    - php: 5.5
```
Comment (ADmad) [inline .travis.yml]: using 7.0 makes 'em run faster :)

File: .travis.yml (hunk 2 above)
Comment (Marlinc) [inline .travis.yml]: The current problem is that CakePHP 3.2 made backwards incompatible changes to the `EntityInterface`. This means that this plugin has to be either CakePHP 3.2 up or specifically require a version lower than 3.2.

Those backwards incompatible changes can be seen at https://github.com/cakephp/cakephp/commit/ad386871f27abcdc35e8d2b170637d01d4d790c2#diff-fcf8477a981a8f104e4d2d3d2b4ffdbfL65 and can be noticed in https://travis-ci.org/UseMuffin/Webservice/jobs/106965645#L319

Comment (ADmad) [inline .travis.yml]: In that case if existing master is stable enough then we should first tag it and then bump version after merging this change while updating core dependency in composer.json

Comment (Marlinc) [inline .travis.yml]: I'd say its quite stable, I think it can be tagged.

Comment (ADmad) [inline .travis.yml]: Okay, @jadb sounds good?

Comment (jadb) [inline .travis.yml]: Yes, it could be tagged.

File: src/Webservice/Webservice.php [hunk 4]
```diff
@@ -282,15 +283,15 @@ protected function _logQuery(Query $query, LoggerInterface $logger)
...
      *
      * @param array $results Array of results from the API
-     * @param string $resourceClass The resource class to use
+     * @param \Muffin\Webservice\Model\Endpoint $endpoint The endpoint class to use
      *
-     * @return Resource[] Array of resource objects
+     * @return \Muffin\Webservice\Model\Resource[] Array of resource objects
      */
-    protected function _transformResults(array $results, $resourceClass)
+    protected function _transformResults(array $results, Endpoint $endpoint)
```
Comment (jadb) [inline src/Webservice/Webservice.php]: I usually prefer object passed first ahead of the rest - if we are to break the API (changing from string to object) should the signature be changed as well?

**Comment by Marlinc** [comment]
I've opened cakephp/cakephp#8211 which should resolve the issue related to _className tests failing.

**Comment by Marlinc** [comment]
I've just tested the patch locally and it appears to work. Waiting for core to merge it. Although I'm sure they need to release a new version first.

**Comment by Marlinc** [comment]
This should not be merged yet.

File: src/Model/Resource.php [hunk 5]
```diff
@@ -4,8 +4,9 @@
 
 use Cake\Datasource\EntityInterface;
 use Cake\Datasource\EntityTrait;
+use Cake\Datasource\InvalidPropertyInterface;
 
-class Resource implements EntityInterface
+class Resource implements EntityInterface, InvalidPropertyInterface
```
Comment (Marlinc) [inline src/Model/Resource.php]: Have to remove this, this makes it dependent on CakePHP 3.2. We'll have to add this once we tag the plugin for 3.1.

**Pull request merged**